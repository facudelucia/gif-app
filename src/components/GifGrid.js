import React, {Fragment} from 'react'
import GifGridItem from './GifGridItem'
import useFetchGifs from "../hooks/useFetchGifs"

const GifGrid = ({ category }) => {
    const {data:images, loading} = useFetchGifs(category)

    return (
        <Fragment>
            <h3 className="animate__animated animate__bounce">{category}</h3>
            {loading && <p className="animate__animated animate__flash">Loading...</p>}
            <div className="card-grid">
                
                    {images.map(imagen => {
                        return <GifGridItem key={imagen.id} imagen={imagen} />
                    })}
                
            </div> 
        </Fragment>
        
     );
}

export default GifGrid;