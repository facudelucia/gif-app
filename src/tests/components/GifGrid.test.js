import React from 'react'
import { shallow } from "enzyme"
import GifGrid from '../../components/GifGrid'
import '@testing-library/jest-dom'
import useFetchGifs from '../../hooks/useFetchGifs'
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas en <GifGrid />', () => {
    const category = "string"

    test('Debe mostrar el componente correctamente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
        let wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot()
    })
    test('Debe mostrar items cuando useFetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://google.com',
            title: 'Cualquier cosa'
        }]
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })
        let wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('p').exists()).toBe(false)
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
    })
})
