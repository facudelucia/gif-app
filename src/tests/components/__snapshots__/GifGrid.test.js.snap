// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    string
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid /> Debe mostrar items cuando useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    string
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://google.com"
    />
  </div>
</Fragment>
`;
